include/vsag/index_feature.h

```diff
@@ -17,8 +17,8 @@
 
 namespace vsag {
 
-enum class IndexFeature {
-    NEED_TRAIN, /**< Indicates that the class needs to be trained */
+enum IndexFeature {
+    NEED_TRAIN = 0, /**< Indicates that the class needs to be trained */
```

Comment:
start from 1, 0 is reserved

Reply:
why reserve 0, I have make the last one is the count of the enum, if start with 1,  when I cast 0 to a IndexFeature type will cause error.

Reply:
reserve of 0 is useful for exposing the problems with uninitiated variables or unexpected type casting

Reply:
done